compose/cli/main.py

```diff
@@ -1312,3 +1321,51 @@ def build_exec_command(options, container_id, command):
     args += [container_id]
     args += command
     return args
+
+
+def has_container_with_state(containers, state):
+    for container in containers:
+        states = {
+            'running': container.is_running,
+            'stopped': not container.is_running,
+            'paused': container.is_paused,
+        }
+        if state not in states:
+            raise UserError("Invalid state: %s" % state)
+        if states[state]:
+            return True
+    return False
+
+
+def filter_services(filters, services, project):
+    def should_include(service):
+        for f in filters:
+            if f == 'status':
+                containers = project.containers([service.name], stopped=True)
+                for status in filters[f]:
+                    if not has_container_with_state(containers, status):
+                        return False
+            elif f == 'key':
+                for key in filters[f]:
+                    if key == 'image' or key == 'build':
+                        if key not in service.options:
+                            return False
+                    else:
+                        raise UserError("Invalid option: %s" % key)
```

Comment:
Maybe `"Invalid value for key filter: %s"`  would be a more useful error message here?

Reply:
Done!